src/python/pants/bsp/types.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
This file seems like a catch-all. Did you consider other places for this class?

Reply:
I thought about putting it rules.py, but that seems overloaded with lots of stuff at the moment. Probably time for a `util_rules` type split, but that should be done in a separate PR.

Any good name for a file with base types in it?

Reply:
> Any good name for a file with base types in it?

No because I generally prefer having functionality located near the class name. So I'm more pro `util_rules` style split. But this is all hand-wavey. It's fine to use this for now, especially given that it's experimental

Reply:
I'll leave it here for now and put it with the init logic when I break that out into a separate file. Probably tomorrow or early next week.